lib/recipe.rb

```diff
@@ -0,0 +1,28 @@
+class Recipe
+  attr_reader :name, :needed
+
+  def initialize(name)
+    @name = name
+    @needed = Hash.new(0)
+  end
+
+  def ingredients_required
```

Comment:
If you would have just named the instance variable @ingredients_requried, then you wouldn't have needed this extra method.